ide/vscode-support/client/src/extension.ts

```diff
@@ -0,0 +1,180 @@
+
+    panel.webview.onDidReceiveMessage(message => {
+        if (message.command === 'enableLSP') {
+            handleConnectionUpdate(message.host, message.port);
+        }
+    }, undefined, extensionContext.subscriptions);
+
+    panel.webview.html = generateWebviewHTML();
+}
+
+function handleConnectionUpdate(host: string, port: string) {
+    writeToTerminal(`Updating connection to Host: ${host}, Port: ${port}`);
+    if (isValidHost(host) && isValidPort(port)) {
+        connectionInfo.host = host;
+        connectionInfo.port = parseInt(port, 10);
+        if (client) {
+            client.stop().then(() => {
+                startLanguageClient(extensionContext);
+            });
+        } else {
+            startLanguageClient(extensionContext);
+        }
+    }
+    vscode.window.showInformationMessage(`Connection updated to Host: ${host}, Port: ${port}`);
+}
+
+function isValidHost(host: string): boolean {
+    return !!host;
+}
+
+function isValidPort(port: string): boolean {
+    const portNumber = parseInt(port, 10);
+    return !isNaN(portNumber) && portNumber > 0 && portNumber < 65536;
+}
+
+function generateWebviewHTML() {
+    return `
+    <html>
+        <head>
+            <style>
+                body {
+                    font-family: Arial, sans-serif;
+                    display: flex;
+                    flex-direction: column;
+                    align-items: center;
+                    justify-content: flex-start;
+                    height: 100vh;
+                    padding-top: 20px;
+                }
+                h1 {
+                    font-size: 24px;
+                    margin-bottom: 20px;
+                }
+                .input-group {
+                    display: flex;
+                    align-items: center;
+                    margin-bottom: 10px;
+                }
+                .input-group input {
+                    margin-right: 10px;
+                }
+            </style>
+            <script>
+                const vscode = acquireVsCodeApi();
+
+                window.onload = () => {
+                    document.getElementById('enableButton').addEventListener('click', () => {
+                        const host = document.getElementById('hostInput').value;
+                        const port = document.getElementById('portInput').value;
+                        vscode.postMessage({ command: 'enableLSP', host, port });
+                    });
+                };
+            </script>
+        </head>
+        <body>
+            <h1>Adama Configuration</h1>
+            <div class="input-group">
+                <input type="text" id="hostInput" placeholder="localhost">
+                <input type="text" id="portInput" placeholder="2423">
+                <button id="enableButton">Enable</button>
+            </div>
+        </body>
+    </html>`;
+}
+
+export function deactivate(): Thenable<void> | undefined {
```

Comment:
as a nit, the tabbing is inconsistent. Let's auto format to have 2 characters as a tab.